webapp/src/lib/Ui.ts

```diff
@@ -70,11 +115,17 @@ export async function showTransactions(dialog: MatDialog, title: string, worker:
 
     return new Promise((resolve) => {
         tc.afterClosed().subscribe((res) => {
-            resolve(res);
+            resolve(res as T);
```

Comment:
Casting, especially in functional world, is indicative of a code smell. Why would you need that (it works if I remove it)?

Reply:
oups - leftover of me getting acquainted with generics...